server/models/Transaction.js

```diff
@@ -27,7 +24,42 @@ export default (Sequelize, DataTypes) => {
       }
     },
 
-    // stores the currency that the transaction happened in (currency of the host)
+    UserId: {
+      type: DataTypes.INTEGER,
+      references: {
+        model: 'Users',
+        key: 'id'
+      },
+      onDelete: 'SET NULL',
+      onUpdate: 'CASCADE',
+      allowNull: false
+    },
+
+    // Keeps a reference to the host because this is where the bank account is
+    // Note that the host can also change over time (that's why just keeping GroupId is not enough)
+    HostId: {
+      type: DataTypes.INTEGER,
+      references: {
+        model: 'Users',
+        key: 'id'
+      },
+      onDelete: 'SET NULL',
+      onUpdate: 'CASCADE',
+      allowNull: false
+    },
+
+    GroupId: {
+      type: DataTypes.INTEGER,
+      references: {
+        model: 'Groups',
+        key: 'id'
+      },
+      onDelete: 'SET NULL',
+      onUpdate: 'CASCADE',
+      allowNull: false
```

Comment:
There are only 3 transactions done by Pia more than a year ago and that already have a `deletedAt`.
I would recommend to delete them so that we can enforce the requirement of `GroupId` NOT NULL.
![](https://cl.ly/1v3227080V3O/Screen%20Shot%202017-06-14%20at%201.49.50%20PM.png)